unity/ComputerVision-UnityClient/Assets/Scripts/AppManager.cs

```diff
@@ -0,0 +1,370 @@
+                        GUI.DrawTexture(new Rect(faceDetectionRects[i][0] * imgScalingFactor, faceDetectionRects[i][1] * imgScalingFactor, width, height), rectTexture, ScaleMode.StretchToFill, true, width / height);
+                    }
+                    break;
+                    
+                case ServiceMode.FaceRecognition:
+                    height = imgScalingFactor* (faceRecognitionRect[3] - faceRecognitionRect[1]);
+                    width = imgScalingFactor * (faceRecognitionRect[2] - faceRecognitionRect[0]);
+                    GUI.DrawTexture(new Rect(faceRecognitionRect[0] * imgScalingFactor, faceRecognitionRect[1] * imgScalingFactor, width, height), rectTexture, ScaleMode.StretchToFill, true, width/height);
+                    TextStyle.normal.textColor = getConfidenceLevelColorFR(faceRecognitionConfidenceLevel);
+                    TextStyle.fontSize = 50;
+                    TextStyle.fontStyle = FontStyle.Bold;
+                    GUI.Label(new Rect(faceRecognitionRect[0] * imgScalingFactor, faceRecognitionRect[1] * imgScalingFactor + 50, width, 100), new GUIContent(faceRecognitionSubject), TextStyle);
+                    break;
+
+                case ServiceMode.ObjectDetection:
+                    foreach (@Object obj in objectsDetected)
+                    {
+                        height = imgScalingFactor * (obj.rect[3] - obj.rect[1]);
+                        width = imgScalingFactor * (obj.rect[2] - obj.rect[0]);
+                        GUI.DrawTexture(new Rect((obj.rect[0] * imgScalingFactor), (obj.rect[1] * imgScalingFactor), width, height), rectTexture, ScaleMode.StretchToFill, true, width / height);
+                        TextStyle.normal.textColor = getConfidenceLevelColorOD(obj.confidence);
+                        TextStyle.fontSize = 30;
+                        TextStyle.fontStyle = FontStyle.Bold;
+                        GUI.Label(new Rect(obj.rect[0] * imgScalingFactor, (obj.rect[1] * imgScalingFactor) + 50, width, 100), new GUIContent(obj.@class), TextStyle);
+                    }
+                    break;
+            }
+
+        }
+        public void ClearGUI()
+        {
+            faceDetectionRects = null;
+            faceRecognitionRect = null;
+            objectsDetected = null;
+        }
+
+        /// <summary>
+        /// Shrinks the screen shot taken to the supplied TargetWidth, then encode the new texture a JPG format and returns the binary array
+        /// Shrinking happens by organizing the pixels of the source img into the new scaled texture using the normalized UV map
+        /// </summary>
+        /// <param name="source">Screen Shot Texture</param>
+        /// <param name="targetWidth"></param>
+        /// <returns> shrank image binary </returns>
+        byte[] ShrinkAndEncode(Texture2D source, int targetWidth)
+        {
+            imgScalingFactor = source.width / targetWidth;
```

Comment:
If targetWidth is smaller than source.width, I think I would make imgScalingFactor = targetWidth / source.width and then multiply instead of divide when scaling. But if your current method makes more sense to you, it looks good

Reply:
Good call but the source.width(Screen Width) in most of the phones will bigger than 240 (I am also enforcing landscape view)